test/pylib/harness_cli.py

```diff
@@ -0,0 +1,168 @@
+        """Before a test starts check if cluster needs cycling and update driver connection"""
+        dirty = await self.is_dirty()
+        if dirty:
+            self.driver_close()  # Close driver connection to old cluster
+        await self._request_and_check(f"http://localhost/cluster/before_test/{test_name}",
+                                      f"Could not get a new cluster for test {test_name}")
+        if self.cql is None:
+            # TODO: if cluster is not up yet due to taking long and HTTP timeout, wait for it
+            # await self._wait_for_cluster()
+            await self.driver_connect()  # Connect driver to new cluster
+
+    async def after_test(self, test_name: str) -> None:
+        """Tell harness this test finished"""
+        await self._request_and_check(f"http://localhost/cluster/after_test/{test_name}",
+                                      f"Could not tell harness test {test_name} ended")
+
+    async def _request_and_check(self, url: str, msg: str, expect: Optional[str]=None) -> str:
+        resp = await self.session.get(url)
+        if resp.status >= 500:
+            raise HarnessOpException(msg)
+        text = await resp.text()
+        if expect is not None:
+            assert text == expect, f"Request to Harness failed, got {text}, expected {expect}"
+        return text
+
+    async def is_harness_up(self) -> bool:
+        """Check if Harness server is up"""
+        ret = await self._request_and_check("http://localhost/up",
+                                            "Could not get if harness is up")
+        return ret == "True"
+
+    async def is_cluster_up(self) -> bool:
+        """Check if cluster is up"""
+        ret = await self._request_and_check("http://localhost/cluster/up",
+                                            "Could not get if cluster is up")
+        return ret == "True"
+
+    async def is_dirty(self) -> bool:
+        """Check if current cluster dirty."""
+        dirty = await self._request_and_check("http://localhost/cluster/is-dirty",
+                                              "Could not check if cluster is dirty")
+        return dirty == "True"
+
+    async def replicas(self) -> int:
+        """Get number of configured replicas for the cluster (replication factor)"""
+        resp = await self._request_and_check("http://localhost/cluster/replicas",
+                                             "Error getting replicas (RF) from cluster")
+        return int(resp)
+
+    async def servers(self) -> List[str]:
+        """Get list of running servers"""
+        host_list = await self._request_and_check("http://localhost/cluster/servers",
+                                                  "Error getting list of servers")
+        return host_list.split(',')
+
+    async def mark_dirty(self) -> None:
+        """Manually mark current cluster dirty.
+           To be used when a server was modified outside of this API."""
+        self.dirty = True
```

Comment:
could you please explain why you need self.dirty on the client? Shouldn't you trust the server?

Reply:
I don't, will remove